docs/source/caching/cache-key-resolution.mdx

```diff
@@ -8,181 +8,66 @@ The normalized cache computes a **cache key** for each object that is stored in
-
- The code generation engine creates this file if it doesn't exist yet, but never overwrites an existing `SchemaConfiguration.swift` file. This means you can edit your schema configuration without those changes being overwritten on subsequent code generation runs.
-
-> *Tip:* You can configure the location of the generated schema types with the [`output.schemaTypes` option in your code generation configuration](./../code-generation/codegen-configuration).
-
-## Specifying cache IDs
-
-The `SchemaConfiguration` contains a [`cacheKeyInfo(for type:object:)`](https://www.apollographql.com/docs/ios/docc/documentation/apolloapi/schemaconfiguration/cachekeyinfo(for:object:)) function. This function provides you a JSON response `object` and the concrete `type` of the object represented by the response.
-
-The `object` parameter provides a JSON response from either a network request or a cache hit.
-
-The `type` parameter provides a strongly typed [`Object`](https://www.apollographql.com/docs/ios/docc/documentation/apolloapi/object) type. This is a generated metadata type representing a concrete `type` in your GraphQL schema.
-
-To configure how cache keys are computed from a response object, you can create and return [`CacheKeyInfo`](#cachekeyinfo) values from your implementation of [`cacheKeyInfo(for:object:)`](https://www.apollographql.com/docs/ios/docc/documentation/apolloapi/schemaconfiguration/cachekeyinfo(for:object:)).
-
-> **Note:** When specifying **cache IDs**, make sure that you are always fetching the fields used to construct those IDs in your operations. Any response objects that don't contain the **cache ID** fields will not be able to be [merged via cache normalization](./introduction#normalizing-responses).
-
-### Using a default cache ID field
-
-If your schema provides a common unique identifier across many of your objects types, you may want to use that field as the **cache ID** by default.
-
-```swift title="SchemaConfiguration.swift"
-public enum SchemaConfiguration: ApolloAPI.SchemaConfiguration {
-  static func cacheKeyInfo(for type: Object, object: ObjectData) -> CacheKeyInfo? {
-    guard let id = object["id"] as? String else {
-        return nil
-    }
-
-    return CacheKeyInfo(id: id)
-  }
-}
+```graphql title="typePolicies.graphqls"
+extend type Book @typePolicy(keyFields: "id")
 ```
 
-If the JSON response `object` has no `id` field, the function returns `nil` and the cache will normalize the object using the [default response path normalization](./introduction#normalizing-responses).
+Apollo iOS will now resolve the cache key for all objects with a `__typename` of `"Book"` by using the value of their `id` field. The concrete type of the object is always prepended to the cache key. This means that a `Book` object with an `id` of `456` will resolve to have a cache key of `Book:456`.
```

Comment:
Might be worth adding that either:

1. it's the user responsibility to make sure all the key fields are queried
2. or that the codegen is doing it automatically

I think it is 1. right now? Either case, it's an important consideration for the system to work.